src/main/java/com/frameworkium/core/ui/pages/BasePage.java

```diff
@@ -100,15 +97,14 @@ public T get(long timeout) {
     @SuppressWarnings("unchecked")
     public T get() {
 
+        //Populate Page Object
         HtmlElementLoader.populatePageObject(this, driver);
 
-        // wait for page to load
-        if (isPageAngularJS()) {
-            waitForAngularRequestsToFinish();
-        }
+        //Wait for Elements & JS
         visibility.waitForAnnotatedElementVisibility(this);
+        javascriptWait.waitForJavascriptEventsOnLoad();
```

Comment:
I originally had it the other way round, but had to switch it as it didn't detect any JS framework as the document was returning ready before Angular 2 was even ready. This was also the same for the Angular documentation unit test which uses Angular 1.

Before 2.0, I'm pretty sure the workflow was the same as above, it checked the elements prior to the Angular wait.